### PR TITLE
x509 certificates for validation

### PR DESCRIPTION
**Pull request by shreyav**

### CODE REVIEW

File: uma-sdk/src/commonMain/kotlin/me/uma/protocol/PubKeyResponse.kt [hunk 1]
```diff
@@ -30,16 +76,37 @@ data class PubKeyResponse @JvmOverloads constructor(
...
+
+private fun X509Certificate.getPubKeyBytes(): ByteArray {
+    if (publicKey !is ECPublicKey ||
+        !(publicKey as ECPublicKey).params.toString().contains("secp256k1")
+    ) {
+        throw IllegalStateException("Public key extracted from certificate is not EC/secp256k1")
+    }
+    // encryptionPubKey.publicKey is an ASN.1/DER encoded X.509/SPKI key, the last 65
+    // bytes are the uncompressed public key
+    return publicKey.encoded.takeLast(65).toByteArray()
```
Comment (jklein24) [inline uma-sdk/src/commonMain/kotlin/me/uma/protocol/PubKeyResponse.kt:111]: Ooh cool :-). This starts with `04` and doesn't include the `---- END BLAH ----` tags, right?

File: javatest/src/test/java/me/uma/javatest/UmaTest.java [hunk 2]
```diff
@@ -252,6 +264,23 @@ callback, new PubKeyResponse(publicKeyBytes(), publicKeyBytes()),
...
+    @Test
+    public void serializeAndDeserializePubKeyResponse() {
+        PubKeyResponse keysOnlyResponse =
+                new PubKeyResponse(UmaTest.hexToBytes("02d5fe"), UmaTest.hexToBytes("123456"));
+        String json = keysOnlyResponse.toJson();
+        PubKeyResponse parsedResponse = umaProtocolHelper.parseAsPubKeyResponse(json);
+        assertNotNull(parsedResponse);
+        assertEquals(keysOnlyResponse, parsedResponse);
+
+        PubKeyResponse certsOnlyResponse =
```
Comment (jklein24) [inline javatest/src/test/java/me/uma/javatest/UmaTest.java:276]: We should also test a response that has all 4 for the backwards compat case.

**Review by jklein24 — APPROVED** [review]

File: uma-sdk/src/commonMain/kotlin/me/uma/protocol/PubKeyResponse.kt (hunk 1 above)
Comment (shreyav) [reply uma-sdk/src/commonMain/kotlin/me/uma/protocol/PubKeyResponse.kt:111]: yes! added an example of what it looks like here https://github.com/uma-universal-money-address/uma-kotlin-sdk/pull/26/files#diff-c3d18805abe410a5e76b66b0d653e58f34ae72a732bb2d5bb85c033f48d98410R30

File: javatest/src/test/java/me/uma/javatest/UmaTest.java (hunk 2 above)
Comment (shreyav) [reply javatest/src/test/java/me/uma/javatest/UmaTest.java:277]: discussed on slack! will handle this alongside backwards compatibility for the other data models

**Pull request merged**